cometd-java/cometd-java-client/src/main/java/org/cometd/client/transport/LongPollingTransport.java

```diff
@@ -138,123 +139,132 @@ public void send(final TransportListener listener, final List<Message.Mutable> m
 
         request.content(new StringContentProvider(generateJSON(messages)));
 
-        customize(request);
-
-        synchronized (this) {
-            if (_aborted) {
-                throw new IllegalStateException("Aborted");
-            }
-            _requests.add(request);
-        }
-
-        request.listener(new Request.Listener.Adapter() {
-            @Override
-            public void onHeaders(Request request) {
-                listener.onSending(messages);
+        customize(request, Promise.from(customizedRequest -> {
+            synchronized (this) {
+                if (_aborted) {
+                    throw new IllegalStateException("Aborted");
+                }
+                _requests.add(customizedRequest);
```

Comment:
Move this `synchronized` block out of the promise and before the call to `customize(Request, Promise)`.

Reply:
Then move the rest of the code below to another method so that the block above becomes something like:

```java
synchronized (this) {
    ...
}
customize(request, Promise.from(customizedRequest -> send2(customizedRequest, ...), failure -> listener.onFailure(failure, messages)));
```